lib/solr.ts

```diff
@@ -907,37 +926,63 @@ export class Client {
     fieldType: string,
     cb: CallbackFn
   ): void {
-    const payload = {
+    const json = JSON.stringify({
       'add-field': {
         name: fieldName,
         type: fieldType,
         multiValued: false,
         stored: true,
       },
+    });
+
+    const callback = (err, result) => {
+      if (err) {
+        // ToDo We should handle this in a more robust way in the future, but
+        // there is a difference between default setup in Solr 5 and Solr 8, so
+        // some fields already exist in Solr 8. Hence if that's the case, we just
+        // ignore that.
+        console.warn(err.message);
+      }
+      cb(undefined, result);
     };
 
-    const params = {
+    const fullPath = `${this.options.path}/${this.options.core}/schema`;
+
+    const requestOptions: RequestOptions = {
       host: this.options.host,
       port: this.options.port,
-      fullPath: `${this.options.path}/${this.options.core}/schema`,
-      json: JSON.stringify(payload),
-      secure: this.options.secure,
-      bigint: this.options.bigint,
-      authorization: this.options.authorization,
-      agent: this.options.agent,
-      ipVersion: this.options.ipVersion,
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json; charset=utf-8',
+        'content-length': Buffer.byteLength(json),
+        accept: 'application/json; charset=utf-8',
+      },
+      path: fullPath,
```

Comment:
same, we replace fullPath with path

Reply:
Ditto. See https://github.com/lbdremy/solr-node-client/pull/278/files/3418634b61cb16b6dca20dad0a8f9d469efa5503#diff-b0d5c84599fa1c42597f5bfad83e01676fdbe2cb87068ed1f57112293b54bd67L108